src/utils/util/resolver-pipeline/testing/4_modified/modify.xspec

```diff
@@ -1,301 +1,2050 @@
+                    <part name="statement" id="new-stmt">
+                        <p>New text</p>
+                    </part>
+                    <p>Text</p>
+                </part>
+            </x:expect>
+        </x:scenario>
+        <x:scenario label="Explicit binding to a descendant that is inside a child control">
+            <x:variable name="ov:add-position" select="'starting'"/>
+            <x:variable name="ov:by-id" select="'a1.1-stmt'"/>
+            <x:like label="SHARED_add-explicit-part-parameterized-position"/>
+            <x:expect label="No change, because the child control's descendants are not in scope for modifications to the parent control">
+                <part name="statement" id="a1.1-stmt">
+                    <p>Text</p>
+                </part>
+            </x:expect>
+            <x:expect label="New text does not appear" test="empty(//p[.='New text'])"/>
+        </x:scenario>
+        <x:scenario label="Explicit binding to a part that a 'remove' element targets for removal, trying to insert content around the removed part (e.g., replacement)">
+            <x:variable name="ov:context">
+                <catalog id="abc">
+                    <control id="a1">
+                        <title>Control A</title>
+                        <prop value="1" name="source1"/>
+                        <prop value="2" name="source2"/>
+                        <part name="statement" id="a1-stmt">
+                            <p>Text</p>
+                        </part>
+                        <part name="statement" id="a1-stmt2">
+                            <p>Text2</p>
+                        </part>
+                    </control>
+                    <modify>
+                        <alter control-id="a1">
+                            <remove by-id="a1-stmt"/>
+                            <add position="after" by-id="a1-stmt">
+                                <part name="statement" id="a1-stmt-after"/>
+                            </add>
+                            <add position="before" by-id="a1-stmt">
+                                <part name="statement" id="a1-stmt-before"/>
+                            </add>
+                        </alter>
+                    </modify>
+                </catalog>
+            </x:variable>
+            <x:context select="$ov:context//*[@id='a1-stmt']">
+                <x:param name="modifications" select="$ov:context//o:modify" tunnel="yes"/>
+            </x:context>
+            <x:expect label="Elements in 'add' that use 'before' and 'after'">
+                <part name="statement" id="a1-stmt-before"/>
+                <part name="statement" id="a1-stmt-after"/>
+            </x:expect>
+            <x:expect label="The target control does not appear" test="empty($x:result//*[@id='a1-stmt'])"/>
+        </x:scenario>
+        <x:scenario label="Explicit binding to a param that needs to be set *and* have new params inserted">
```

Comment:
@wendellpiez : Does this test scenario cover the situation in issue #1859?

Reply:
I don't think so ... what is being described is wanting to add a new `prop` to an existing `param`, but not being able to use `alter`. Apparently `alter` is a no-op against params. `set-param` is the available workaround (it can be used to add new content). But the spec does not say that you can't target `param` elements using `alter/add`, so that should also work.

Reply:
Thanks. You're right, this scenario isn't the same as what the issue is about. But I think the XSLT in this PR fixes the issue. If I create a similar test scenario with
```
<alter control-id="a1">
    <add position="starting" by-id="a1_prm1">
        <prop ns="https://fedramp.gov/ns/oscal" name="param-label" value="AC-1(a)"/>
    </add>
</alter>
```
then I can see that the new `prop` gets added at the beginning of the existing `param`, like this:
```
<param id="a1_prm1">
    <prop ns="https://fedramp.gov/ns/oscal" name="param-label" value="AC-1(a)"/>
    <label>A1 Parameter 1</label>
    <constraint>at least every 3 years</constraint>
</param>
```
Should I add the new test scenario?

Reply:
Sure! I guess I was not looking closely enough, or not looking at the current code, to see that this should actually work.

If it actually works, so much the better, we only need to be able to show that it does.

Reply:
I pushed commit 5d3b9b302d2d06080afeee5eb7a6aa6d176cfaf2 to this PR branch, to provide a test of the situation in #1859. I also tweaked the label of the other test scenario to distinguish the two scenarios from each other more.